_posts/diary/2025-11-23-opensearch-for-static-site-logs.md

```
@@ -0,0 +1,157 @@
+
+I run services on my home server as specific users, not as root or random UIDs. OpenSearch runs as UID 1000 inside its container. UID 1000 on my system is a real user who shouldn't have access to OpenSearch data. The solution seemed obvious:
+
+```yaml
+services:
+  opensearch:
+    user: "1003:1004"  # opensearch user
+```
+
+The containers started, then immediately failed:
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)